ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/validator/preexecution/PreExecutingSubmissionValidatorSpec.scala

```diff
@@ -34,10 +34,45 @@ import org.scalatest.wordspec.AsyncWordSpec
 import scala.collection.JavaConverters._
 import scala.concurrent.{ExecutionContext, Future}
 
-class PreExecutingSubmissionValidatorSpec extends AsyncWordSpec with Matchers with MockitoSugar {
+class PreExecutingSubmissionValidatorSpec
+    extends AsyncWordSpec
+    with Matchers
+    with MockitoSugar
+    with ArgumentMatchersSugar {
   private implicit val loggingContext: LoggingContext = LoggingContext.ForTesting
 
   "validate" should {
+
+    "read from ledger state only once in presence of (KV) state key inputs of DAML contract keys'" in {
+
+      // Given a KV state key input of a DAML contract key, de-referencing it at validation time
```

Comment:
Please move this comment somewhere else or remove it -- it is confusing to have such context here in a unit test.

Reply:
I think it's important background to keep. It is here because it explains the reason of this test; I could move it to the implementation but the logic there has been removed, so it would not be clear (and I'd rather avoid time references such as "once, we were reading twice here..."). Do you have suggestions about where to move it?

Reply:
Then it's better to remove this test case altogether as it does not help understanding how the code works *currently*. This test had (or rather, would have had) a valid existence before we removed the two-phase retrieval of inputs.

Reply:
That test was probably [this one](https://github.com/digital-asset/daml/commit/0ee2159218eda085a084ed07ca2d7b245fbb6c8e#diff-2a635860c9f1ce61e8da3014db84dc0358a0d3e8d6da18e3a92d64f6b3935906R96), @SamirTalwar-DA might confirm/rectify. If you think there's no chance we'll have a similar regression in the future I can remove it.

Reply:
Let's purge this test altogether. I don't believe we need a test case to verify we're not doing something.

Reply:
(And let's delete the other one too.)

Reply:
Done and added performance-related usage advice ScalaDoc to `StateReader.read`.